pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -471,3 +489,41 @@ func (s *Server) getRootCertFromSecret(name, namespace string) (*istioCredential
 	}
 	return kube.ExtractRoot(secret.Data)
 }
+
+// checkAndRunNonRevisionLeaderElectionIfRequired checks the ConfigMap for leader election and runs non-revision leader election
+// to figure out when the older leader has released the lock.
+// Once the lock is acquired, it will close the activateCh channel to signal the per-revision leader to start writing status.
+func (s *Server) checkAndRunNonRevisionLeaderElectionIfRequired(args *PilotArgs, activateCh chan struct{}) {
+	cm, err := s.kubeClient.Kube().CoreV1().ConfigMaps(args.Namespace).Get(context.Background(), leaderelection.GatewayStatusController, v1.GetOptions{})
+	if err != nil {
+		// ConfigMap does not exist, so per-revision leader election should be active
+		close(activateCh)
+		return
+	}
+	leaderAnn, ok := cm.Annotations[k8sresourcelock.LeaderElectionRecordAnnotationKey]
+	if ok {
+		var leaderInfo struct {
+			HolderIdentity string `json:"holderIdentity"`
+		}
+		if err := json.Unmarshal([]byte(leaderAnn), &leaderInfo); err == nil {
+			if leaderInfo.HolderIdentity != "" {
+				// Non-revision leader election should run, per-revision should be waiting for activation
+				s.addTerminatingStartFunc("gateway status", func(stop <-chan struct{}) error {
+					leaderelection.
+						NewLeaderElection(args.Namespace, args.PodName, leaderelection.GatewayStatusController, args.Revision, s.kubeClient).
+						AddRunFunction(func(leaderStop <-chan struct{}) {
+							// now that we have the leader lock, we can activate the per-revision status writer
+							close(activateCh)
```

Comment:
Testing this out I got

```
panic: close of closed channel

goroutine 1294 [running]:
istio.io/istio/pilot/pkg/bootstrap.(*Server).checkAndRunNonRevisionLeaderElectionIfRequired.func1.1(0xc00169fa90?)
        istio.io/istio/pilot/pkg/bootstrap/configcontroller.go:516 +0x17
created by istio.io/istio/pilot/pkg/leaderelection.(*LeaderElection).create.func1 in goroutine 1385
        istio.io/istio/pilot/pkg/leaderelection/leaderelection.go:146 +0xa5
```

Reply:
Because of the crash, I cannot reproduce this but a bug I think exists:

1. We have revision 'old' (default) and 'new'.
2. New wins the election due to old not running for sometime. It starts the per-revision lock
3. Old starts, and since its the default it **steals** the lock and starts the global lock
4. Both are running

Reply:
I'll look into the gorouting panic.  As for the bug, since these two instances are part of two different revisions, this is not a problem, as they will write status only to gateways of their own revision.